cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -49,17 +51,7 @@ const (
 	certmanagerVersionAnnotation = "certmanager.clusterctl.cluster.x-k8s.io/version"
 	certmanagerHashAnnotation    = "certmanager.clusterctl.cluster.x-k8s.io/hash"
 
-	// NOTE: // If the cert-manager.yaml asset is modified, this line **MUST** be updated
-	// accordingly else future upgrades of the cert-manager component will not
-	// be possible, as there'll be no record of the version installed.
-	embeddedCertManagerManifestVersion = "v0.16.1"
-
-	// NOTE: The hash is used to ensure that when the cert-manager.yaml file is updated,
-	// the version number marker here is _also_ updated.
-	// You can either generate the SHA256 hash of the file, or alternatively
-	// run `go test` against this package. THe Test_VersionMarkerUpToDate will output
-	// the expected hash if it does not match the hash here.
-	embeddedCertManagerManifestHash = "af8c08a8eb65d102ba98889a89f4ad1d3db5d302edb5b8f8f3e69bb992faa211"
+	certmanagerVersionLabel = "helm.sh/chart"
```

Comment:
Just wanted to point out that after version v0.16.1 the cert-manager CRDs do not have this label.
So that would mean that checking just `CustomResourceDefinition`s won't work. But maybe that's fine if we don't plan to support cert-manager upgrades to v1.x.x
The label is available on `Deployment` types (if installed via helm charts).

Reply:
thanks for the sharing, do you think we should move this check to deployment then? 
or let it be ? because lack of further info on 0.16.x -> 1.x.x impact and implication of that..

Reply:
Hmm...I'm not fully sure. Maybe @fabriziopandini can chime in. 
My thoughts for now is to keep it as is because currently we do not seem to have a defined plan for upgrading to cert-manager v1.x.x. 
If we do want to do that upgrade, we can fix this as part of that story/issue.

Reply:
thanks, I will keep it for now and hopefully someone else can provide additional comments